util/aggregation.py

```diff
@@ -146,29 +146,34 @@ def aggregate_provenance(job_dir, output_dir, request_id=None):
             json.dump(aggregate_dict, fh, indent=2)
 
 
+def aggregate_annotation_group(job_dir, files):
+    aggregate_dict = {'annotations': []}
+    for f in sorted(files):
+        path = os.path.join(job_dir, f)
+        data = json.load(open(path))
+        for key in data:
+            if key == 'annotations':
+                annotation_list = aggregate_dict[key]
+                new_annotations = [x for x in data[key] if x not in annotation_list]
+                annotation_list.extend(new_annotations)
+            else:
+                aggregate_dict.setdefault(key, {})[f] = data[key]
```

Comment:
aggregate_dict is a dictionary whose values are lists and its setdefault reportedly returns a value if the supplied key is found (a populated list), otherwise the default value (an empty dictionary). If I'm reading this right, I'd say the setdefault's 2nd arg should be an empty list.

Reply:
Actually, aggregate_dict is a dictionary representing JSON. I copied that default assignment from the aggregate_provenance function. In the case of annotation records (with key 'annotations'), we are not concerned about which source file a given annotation came from (and it may have been in several). But for other data (which would have a key other than 'annotations'), we may want to retain that info. So we put other data into aggregate_dict in a structure like that below. Currently, there shouldn't be any keys other than 'annotations'.

{
  'annotations': [<JSON annotation records>],
  'key_1': {
                 'filename1': <JSON value for 'key_1' as loaded from file 'filename1'>,
                 'filename2': <JSON value for 'key_1' as loaded from file 'filename2'>,
                 ...
  },
  ...
}

Reply:
I'll add comments to lines 158 and 159.